[2025-04-21T16:55:53.599-0400] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: breast_cancer_weekly_retrain.retrain_model manual__2025-04-21T20:45:23.115087+00:00 [queued]>
[2025-04-21T16:55:53.603-0400] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: breast_cancer_weekly_retrain.retrain_model manual__2025-04-21T20:45:23.115087+00:00 [queued]>
[2025-04-21T16:55:53.604-0400] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-04-21T16:55:53.612-0400] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): retrain_model> on 2025-04-21 20:45:23.115087+00:00
[2025-04-21T16:55:53.632-0400] {standard_task_runner.py:60} INFO - Started process 84633 to run task
[2025-04-21T16:55:53.639-0400] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'breast_cancer_weekly_retrain', 'retrain_model', 'manual__2025-04-21T20:45:23.115087+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/breast_cancer_retrain_dag.py', '--cfg-path', '/var/folders/xh/nzdhwdsn5tl09sb1dpsfz0r40000gn/T/tmpbp1hrq17']
[2025-04-21T16:55:53.642-0400] {standard_task_runner.py:88} INFO - Job 10: Subtask retrain_model
[2025-04-21T16:55:53.674-0400] {task_command.py:423} INFO - Running <TaskInstance: breast_cancer_weekly_retrain.retrain_model manual__2025-04-21T20:45:23.115087+00:00 [running]> on host gauravs-macbook-pro.local
[2025-04-21T16:55:53.709-0400] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='gaurav' AIRFLOW_CTX_DAG_ID='breast_cancer_weekly_retrain' AIRFLOW_CTX_TASK_ID='retrain_model' AIRFLOW_CTX_EXECUTION_DATE='2025-04-21T20:45:23.115087+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-21T20:45:23.115087+00:00'
[2025-04-21T16:55:53.711-0400] {subprocess.py:63} INFO - Tmp dir root location: /var/folders/xh/nzdhwdsn5tl09sb1dpsfz0r40000gn/T
[2025-04-21T16:55:53.712-0400] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n        source /Users/gaurav/Documents/Breast-Cancer-ML-Pipeline/venv/bin/activate &&\n        cd /Users/gaurav/Documents/Breast-Cancer-ML-Pipeline &&\n        python train.py\n        ']
[2025-04-21T16:55:53.729-0400] {subprocess.py:86} INFO - Output:
[2025-04-21T16:56:03.763-0400] {local_task_job_runner.py:302} WARNING - State of this instance has been externally set to None. Terminating instance.
[2025-04-21T16:56:03.772-0400] {process_utils.py:131} INFO - Sending Signals.SIGTERM to group 84633. PIDs of all processes in the group: [84634, 84635, 84637, 84638, 84639, 84640, 84641, 84642, 84643, 84644, 84645, 84646, 84633]
[2025-04-21T16:56:03.772-0400] {process_utils.py:86} INFO - Sending the signal Signals.SIGTERM to group 84633
[2025-04-21T16:56:03.792-0400] {taskinstance.py:2450} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-04-21T16:56:03.795-0400] {subprocess.py:104} INFO - Sending SIGTERM signal to process group
[2025-04-21T16:56:03.848-0400] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/gaurav/Documents/Breast-Cancer-ML-Pipeline/venv/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/Users/gaurav/Documents/Breast-Cancer-ML-Pipeline/venv/lib/python3.9/site-packages/airflow/operators/bash.py", line 203, in execute
    result = self.subprocess_hook.run_command(
  File "/Users/gaurav/Documents/Breast-Cancer-ML-Pipeline/venv/lib/python3.9/site-packages/airflow/hooks/subprocess.py", line 91, in run_command
    for raw_line in iter(self.sub_process.stdout.readline, b""):
  File "/Users/gaurav/Documents/Breast-Cancer-ML-Pipeline/venv/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 2452, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2025-04-21T16:56:03.854-0400] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=breast_cancer_weekly_retrain, task_id=retrain_model, execution_date=20250421T204523, start_date=20250421T205553, end_date=20250421T205603
[2025-04-21T16:56:03.857-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84644, status='terminated', started='16:55:57') (84644) terminated with exit code None
[2025-04-21T16:56:03.857-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84639, status='terminated', started='16:55:57') (84639) terminated with exit code None
[2025-04-21T16:56:03.858-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84640, status='terminated', started='16:55:57') (84640) terminated with exit code None
[2025-04-21T16:56:03.858-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84646, status='terminated', started='16:55:57') (84646) terminated with exit code None
[2025-04-21T16:56:03.858-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84641, status='terminated', started='16:55:57') (84641) terminated with exit code None
[2025-04-21T16:56:03.858-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84642, status='terminated', started='16:55:57') (84642) terminated with exit code None
[2025-04-21T16:56:03.863-0400] {standard_task_runner.py:107} ERROR - Failed to execute job 10 for task retrain_model (Task received SIGTERM signal; 84633)
[2025-04-21T16:56:03.874-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84634, status='terminated', started='16:55:53') (84634) terminated with exit code None
[2025-04-21T16:56:03.875-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84638, status='terminated', started='16:55:57') (84638) terminated with exit code None
[2025-04-21T16:56:03.875-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84635, status='terminated', started='16:55:53') (84635) terminated with exit code None
[2025-04-21T16:56:03.875-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84637, status='terminated', started='16:55:56') (84637) terminated with exit code None
[2025-04-21T16:56:03.875-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84645, status='terminated', started='16:55:57') (84645) terminated with exit code None
[2025-04-21T16:56:03.875-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84633, status='terminated', exitcode=1, started='16:55:53') (84633) terminated with exit code 1
[2025-04-21T16:56:03.875-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=84643, status='terminated', started='16:55:57') (84643) terminated with exit code None
